phpcs.xml.dist

```diff
@@ -14,18 +14,67 @@
     <file>tests</file>
 
     <rule ref="Doctrine">
+        <!--
+            Remove this rule when using doctrine/coding-standard 8.0.0,
+            which should contain https://github.com/doctrine/coding-standard/pull/169
+        -->
+        <exclude name="SlevomatCodingStandard.ControlStructures.JumpStatementsSpacing.IncorrectLinesCountAfterLastControlStructure"/>
         <exclude name="SlevomatCodingStandard.TypeHints.DeclareStrictTypes"/>
         <exclude name="SlevomatCodingStandard.Classes.SuperfluousAbstractClassNaming"/>
         <exclude name="SlevomatCodingStandard.Classes.SuperfluousExceptionNaming"/>
         <exclude name="SlevomatCodingStandard.ControlStructures.ControlStructureSpacing.IncorrectLinesCountAfterLastControlStructure"/>
         <exclude name="SlevomatCodingStandard.Classes.DisallowLateStaticBindingForConstants.DisallowedLateStaticBindingForConstant"/>
+        <!-- See https://github.com/squizlabs/PHP_CodeSniffer/issues/2937 -->
+        <exclude name="Squiz.Arrays.ArrayDeclaration.ValueNoNewline"/>
     </rule>
 
-    <rule ref="SlevomatCodingStandard.TypeHints.TypeHintDeclaration.MissingParameterTypeHint">
+    <rule ref="Squiz.NamingConventions.ValidVariableName.MemberNotCamelCaps">
+        <!-- This test contains a fixture class that should use snake case  -->
+        <exclude-pattern>tests/Doctrine/Tests/DBAL/Functional/DataAccessTest.php</exclude-pattern>
```

Comment:
Apparently, the CS is already updated in `master` (#3847), and this issue is addressed in a slightly different way.

In order to avoid unnecessary merge conflicts, what if we restart by backporting the CS update from master and applying all the necessary changes on top?

Reply:
Ok, I'll cherry-pick a84752c4382a23ea34cec1cfcfb883417b1cac47, then 78546ce9be08752d9ea2e621b9df816098969e3b (that should result in just an extra rule),  . I will not cherry-pick the commit about automated fixes and will instead run `/vendor/bin/phpcbf`, and finally I will cherry-pick a92fa15198aac398068deb81d331cd79f4f9bb1c

Reply:
> that should result in just an extra rule

It actually doesn't, since some line breaks are already missing on 2.10.x